src/util/psbt/serialize.rs

```diff
@@ -20,3 +26,62 @@ pub trait Deserialize: Sized {
 }
 
 impl_psbt_de_serialize!(Transaction);
+
+impl Serialize for Script {
+    fn serialize(&self) -> Vec<u8> {
+        self.to_bytes()
+    }
+}
+
+impl Deserialize for Script {
+    fn deserialize(bytes: &Vec<u8>) -> Result<Self, encode::Error> {
+        Ok(Self::from(bytes.clone()))
+    }
+}
+
+impl Serialize for PublicKey {
+    fn serialize(&self) -> Vec<u8> {
+        self.serialize().to_vec()
+    }
+}
+
+impl Deserialize for PublicKey {
+    fn deserialize(bytes: &Vec<u8>) -> Result<Self, encode::Error> {
+        let ref mut d = Cursor::new(&bytes[..]);
+
+        let serialized: [u8; 33] = Decodable::consensus_decode(d)?;
```

Comment:
Why go through `consensus_decode`? It's just a passthrough for fixed-size arrays.

Also, you need to check for trailing bytes.